buf/registry/module/v1beta1/branch.proto

```diff
@@ -75,35 +83,41 @@ message Branch {
 message BranchRef {
   option (buf.registry.priv.extension.v1beta1.message).request_only = true;
 
-  oneof value {
-    option (buf.validate.oneof).required = true;
-    // The id of the Branch.
-    string id = 1 [(buf.validate.field).string.uuid = true];
-    // The fully-qualified name of the Branch.
-    BranchFullName name = 2;
+  // The fully-qualified name of a Branch within a BSR instance.
+  //
+  // A Name uniquely identifies a Branch within a BSR instance.
+  // This is used for requests when a caller only has the branch name and not the ID.
+  message Name {
```

Comment:
(nit: might "read" better to have Branch.Name instead of BranchRef.Name, which makes me wonder about Branch.Ref)

Reply:
(which then also makes me wonder about useing Branch.Name inside Branch, instead of flattening the fields into the main message)

Reply:
You can't flatten them because it's in a oneof

Reply:
(I was suggesting unflattening the owner_id, etc, but now I realize those are ids not names)